src/compiler/crystal/syntax/to_s.cr

```diff
@@ -499,9 +499,9 @@ module Crystal
       @str << '`'
       case exp
       when StringLiteral
-        @str << exp.value.inspect[1..-2]
+        @str << exp.value.inspect_unquoted.gsub('`', "\\`")
```

Comment:
`inspect_unquoted` will escape `\"` which is not wrong but also not needed here. Not sure if this should be changed in this PR as well.

Reply:
Yes. backtick operator must need escaping backtick and null character, but it also accepts string escape sequence, so I use `inspect_unquoted`.